### ISSUE TITLE
Tab key behaviour in terminal

### ISSUE THREAD

**Issue by snirye** [microsoft/vscode#182848]
Type: <b>Bug</b>

instead of typing a new TAB, it switch the active frame (moving from the file, to terminal, etc.)
same when trying to accept copilot suggestion.

VS Code version: Code - Insiders 1.79.0-insider (2575777b9d3cdb865d61f9f1dfaac1fea9e1bde6, 2023-05-12T05:21:58.641Z)
OS version: Linux x64 5.19.0-41-generic
Modes:
Sandboxed: Yes
Remote OS version: Linux x64 5.19.0-41-generic
Remote OS version: Linux x64 5.19.0-41-generic
Remote OS version: Linux x64 5.19.0-41-generic
Remote OS version: Linux x64 5.19.0-41-generic
Remote OS version: Linux x64 5.19.0-41-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 4300)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|1, 2, 2|
|Memory (System)|31.08GB (17.17GB free)|
|Process Argv|--unity-launch --crash-reporter-id 58d834bd-61b9-477f-b650-1af72cded714|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|

|Item|Value|
|---|---|
|Remote|Dev Container|
|OS|Linux x64 5.19.0-41-generic|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 4299)|
|Memory (System)|31.08GB (17.17GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|Dev Container|
|OS|Linux x64 5.19.0-41-generic|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 4299)|
|Memory (System)|31.08GB (17.17GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|Dev Container|
|OS|Linux x64 5.19.0-41-generic|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 4299)|
|Memory (System)|31.08GB (17.17GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|Dev Container|
|OS|Linux x64 5.19.0-41-generic|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 4299)|
|Memory (System)|31.08GB (17.17GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|Dev Container|
|OS|Linux x64 5.19.0-41-generic|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 4108)|
|Memory (System)|31.08GB (17.21GB free)|
|VM|0%|
</details><details><summary>Extensions (14)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|13.3.1
vscode-graphql-syntax|Gra|1.1.0
vscode-peacock|joh|4.2.2
jupyter-keymap|ms-|1.1.1
remote-containers|ms-|0.293.0
gitlens|eam|13.6.0
copilot-chat|Git|0.1.2023051701
copilot-nightly|Git|1.86.102
vscode-pull-request-github|Git|0.64.0
go|gol|0.38.0
rest-client|hum|0.25.1
vscode-docker|ms-|1.25.1
vscode-icons|vsc|12.4.0
vscode-proto3|zxh|0.5.5

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256198
vslsvsres303:30308271
pythontb:30258533
vsc_aacf:30263846
pythonptprofiler:30281269
vsdfh931cf:30280410
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30404738
cppdebug:30492333
vsclangdf:30492506
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30618038
pylantcb52:30590116
pyind779:30611226
pythonsymbol12:30651887
showlangstatbar:30737417
azdwalk:30721579
pythonms35:30671666
a2ce3375:30680469
7ij38806:30687742
pythonfmttext:30716741
pythoncmvfstr:30726892
fixhidewlkth:30724384
showsbindicator:30724477
pythongtdpath:30726887

```

</details>

**Comment by gjsjohnmurray** [comment]
Does your status bar show this?

![image](https://github.com/microsoft/vscode/assets/6726799/bae6813e-32c5-469a-912e-e0d15d0473b0)

If so, click on it to turn the feature off. I guess you inadvertently pressed the keyboard shortcut for the command `Toggle Tab Key Moves Focus`.

**Comment by apastel** [comment]
I'm having this issue, but "Tab Moves Focus" is turning itself back on as soon as I click back inside the terminal.

![Peek 2023-05-24 14-54](https://github.com/microsoft/vscode/assets/8836343/7ae215f8-4ef0-4ebc-a65d-ea1de2357c04)

**Comment by m1keil** [comment]
I can confirm the same thing happens to me. The feature auto turns on for some reason. Also, hitting the button at the bottom doesn't disable it, instead I need to use the Ctrl+Cmd+M shortcut. This started happening with the latest 1.81 release.

**Comment by Tyriar** [comment]
@meganrogge confusion around the terminal tab moves focus mode. I forgot where we landed on this, but I would expect it to not do anything unless the user explicitly changes it in settings.

**Comment by meganrogge** [comment]
we landed on respecting the last toggled state for each feature per session. we can rework this but this is behaving as designed

**Comment by m1keil** [comment]
I think I figured out how I ended up in the state where tab focus mode for terminal kept being enabled after re-opening vscode:

1. Enable `Editor: Tab focus mode` 
2. Reset to default `Terminal > Integrated: Tab Focus Mode`
3. Disable `Editor: Tab focus mode` 

Result: 
`terminal.integrated.tabFocusMode` will remain `true` in settings.json.

**Comment by meganrogge** [comment]
Hm, thanks but those steps do not reproduce an issue for me.. 🤔

**Comment by meganrogge** [comment]
I feel like we should go back to just having one that can be toggled and is always in sync / updating the setting value.

**Comment by m1keil** [comment]
Hm, sorry, not sure what makes my config unique in that case. I noticed that even when I reset the `Terminal > Integrated: Tab Focus Mode` setting to default, I still get the "default setting was changed" highlight. So maybe something else is dirty in my config:

![CleanShot_20230720T175719Z@2x](https://github.com/microsoft/vscode/assets/1589008/a83d4887-19e1-445c-afcd-5687fae941a8)

But I'm getting the feeling that this is not a wide bug but just something unique to me and maybe few others. Thanks for trying to reproduce!

**Comment by meganrogge** [comment]
This should be fixed as we no longer have two separate modes (workbench vs terminal)
